### DISCUSSION TITLE
Command Bar Modes

### DISCUSSION THREAD

**Discussion by franknoirot** [KittyCAD/modeling-app#1468] [Milestones]
We've been discussing a bit about how to have _everything_ available in the command bar (which you can open with `Cmd + K` on Mac, and `Ctrl + K` on Windows and Linux), but not make the command bar feel crowded or chaotic.

I suggest that we add **modes** to the command bar that change what you are searching for. These modes will be able to be entered one of two ways, borrowing from the examples set by VS Code and GitHub's command bars:
1. Via similar but modified keyboard shortcut, ex. `Cmd + Shift + K` instead of `Cmd + K`
2. Via a leader character being entered into the command bar (or deleted from it)

Our team seems to agree that this is a useful pattern, but I want to allow space for discussing a couple specifics:
1. What are all the modes we want to support for now?
2. What is the default mode?
3. What are the leader keys and characters that activate the non-default modes?

**Comment by franknoirot** [comment]
Here's my proposal:
1. Default mode - `Mod + K`
  - In home, searches File and Settings commands (like now)
  - In a project, searches Modeling and File commands
2. File Search mode - `Mod + Shift + K` or leader character `/` (lol I just realized Github's file search mode is `/`)
  - In home, searches across all projects
  - In a project, searches only that project (should we support searching other projects with a Github-like `!` project search mode?)
3. Everything Bagel mode - `Mod + Alt + K` or leader character `+`
  - In home, has no effect (that I can think of)
  - In a project, searches all commands: modeling, file, settings, and everything
4. Help mode - `Mod + ?` or leader character `?`
  - Inspired by tips and help leader character from GitHub's command bar
  <img src="https://github.com/KittyCAD/modeling-app/assets/23481541/d96e3350-1bc4-42c5-b978-0c43164d97fe" width="480" />

  - See keyboard shortcuts, tips, command templates, and onboarding guides

**Reply by Irev-Dev** [reply]
Sounds good to me !

**Reply by franknoirot** [reply]
Just realized I never got around to mocking this up really. [There is a mockup in Figma here](https://www.figma.com/design/5UUPOawPLEs3D0Gptf42vy/Designs?node-id=1678-44336&t=O90zEetfmbW6BtJn-11) that shows a "footer" added to the command selection step but that's just a trivial tweak, not the real mode changes.

![mockup showing the command palette with a footer guiding users how to navigate the command selection step](https://github.com/user-attachments/assets/b4ec8c35-523a-49f3-aaab-cbfb03d4a3de)
